users/views.py

```diff
@@ -123,6 +123,19 @@ def get_context_data(self, **kwargs):
 class TunnistamoOidcAuthorizeView(AuthorizeView):
     def get(self, request, *args, **kwargs):
         request.GET = _extend_scope_in_query_params(request.GET)
+
+        if request.GET.get('client_id'):
+            try:
+                client = Client.objects.get(client_id=request.GET.get('client_id'))
```

Comment:
Well, I was too slow to comment. I guess the only point that's important is that validation happens before anything is done with the data. If the validation already happens in the super().get -call, could we just move the line where the id is save to `request.session` after the supe().get -call?

Reply:
Something like this?

```python
class TunnistamoOidcAuthorizeView(AuthorizeView):
    def get(self, request, *args, **kwargs):
        request.GET = _extend_scope_in_query_params(request.GET)

        request_locales = [l.strip() for l in request.GET.get('ui_locales', '').split(' ') if l]
        available_locales = [l[0] for l in settings.LANGUAGES]
        
        response = None
        for locale in request_locales:
            if locale in available_locales:
                with translation.override(locale):
                    response = super().get(request, *args, **kwargs)
                    break

        if not response:
            response = super().get(request, *args, **kwargs)
        
        if request.GET.get('client_id'):
            # Save the client_id to the session to be used in the HelsinkiTunnistus
            # social auth backend.
            request.session["oidc_authorize_original_client_id"] = request.GET.get('client_id')

        return response
```

Maybe could move that locale thing to its own method...

Reply:
Yep, that looks good!

Reply:
I just realised that a client error ends up in the response as a rendered html... Would need to detect that somehow. I think the manual client checking is the simplest, all things considered.

Reply:
Oh yes, true. Approved!